src/Commands/KeyVault/Secret/SecretCreateCommand.cs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.CommandLine;
+using System.CommandLine.Parsing;
+using Azure.Security.KeyVault.Secrets;
+using AzureMcp.Arguments.KeyVault.Secret;
+using AzureMcp.Models.Argument;
+using AzureMcp.Models.Command;
+using AzureMcp.Services.Interfaces;
+using Microsoft.Extensions.Logging;
+using ModelContextProtocol.Server;
+
+namespace AzureMcp.Commands.KeyVault.Secret;
+
+public sealed class SecretCreateCommand(ILogger<SecretCreateCommand> logger) : SubscriptionCommand<SecretCreateArguments>
+{
+    private readonly ILogger<SecretCreateCommand> _logger = logger;
+    private readonly Option<string> _vaultOption = ArgumentDefinitions.KeyVault.VaultName.ToOption();
+    private readonly Option<string> _secretNameOption = ArgumentDefinitions.KeyVault.SecretName.ToOption();
+    private readonly Option<string> _secretValueOption = ArgumentDefinitions.KeyVault.SecretValue.ToOption();
+
+    protected override string GetCommandName() => "create";
+
+    protected override string GetCommandDescription() =>
+        """
+        Create a new secret in an Azure Key Vault. This command creates a secret with the specified name and value
+        in the given vault.
+
+        Required arguments:
+        - subscription
+        - vault-name
+        - secret-name
+        - secret-value
```

Comment:
Do we need these?  I think we can put required in the argumentdefinitions to communicate that to the client